operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java

```diff
@@ -246,4 +259,154 @@ public void testHostnameStrictDisabled() {
         }
     }
 
+    @Test
+    public void testConfigSecretsAreWatched() {
```

Comment:
This test is pretty complicated, if something changes it's hard to maintain, I do believe that it can be split up in 2/3 separate tests.

I think we have a separate story for checking multiple deployments in the same namespace, I would be ok if we just properly test one deployment for now.

Last but not least, in this test I miss to see that we are checking the changed values in the secrets are actually propagated to the instances (am I missing something?) and this is even more important to verify than the rolling update, I guess.

Reply:
As discussed offline, the reasoning was to optimize the tests as each KC deployment is very expensive. But I agree it's not very ideal to have it as a monolithic test from the code maintenance PoV.

Will do tomorrow. 👍